### PR TITLE
Fix displaying edited amount in IOU preview for scan failed receipt

### PR DESCRIPTION
**Pull request by shahinyan11**
### Fixed Issues

$ https://github.com/Expensify/App/issues/34418
PROPOSAL: https://github.com/Expensify/App/issues/34418#issuecomment-1888883374

### Tests

1. Open New Expensify app.
2. Log in as the WS employee
3. Create fail scan request
4. After the scanned receipt shows a failure
5. Navigate to the workspace chat
6. Click on the preview to navigate to the report conversation
7. Edit the missing field Amount only
8. Navigate back to the IOU preview
9. Verify that the edited amount is visible in IOU preview

- [x] Verify that no errors appear in the JS console

### Offline tests

N/A

### QA Steps

Same as in the Tests section.
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/46921547/572ff276-4f66-4ae9-bd0c-7fc4a3266780

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/46921547/2625cfb6-463f-4cf8-bc94-d3daf7106992

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/46921547/af99a303-a2fd-45f9-a28a-723662724d7c

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/46921547/36409b4b-dc8d-4875-a375-c9d7e14a7f2f

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/46921547/7647a999-d096-48c9-9bad-6f932f53a549

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/46921547/3681cd9d-e8a2-4181-80d1-1906a33bc63b

</details>

### CODE REVIEW

File: src/components/ReportActionItem/MoneyRequestPreview.tsx
Comment (shahinyan11) [reply src/components/ReportActionItem/MoneyRequestPreview.tsx]: Yes. No problem

**Comment by shahinyan11** [comment]
> Can we just get a quick confirmation from @Expensify/design wether the layout here looks ok. We're just enabling showing the amount along with the `Receipt missing details` error:

Should I do something about this ? Or you will do it

**Comment by dannymcclain** [comment]
Something about that "Receipt missing details" text doesn't look right to me. @JmillsExpensify @trjExpensify is that how that message is supposed to show?

Other violations get displayed a bit differently (see screenshot) but I'm not sure if this is really the same thing.

CleanShot 2024-02-14 at 08 27 00@2x

**Comment by trjExpensify** [comment]
Yep, agree with Danny. It would be best to put that in the dot separator placement. If there's a merchant or description added, it would go where you've currently got `Receipt missing details`.

**Comment by trjExpensify** [comment]
I think the pattern does exist, just only on paid workspace plans that have violations. CC: @JmillsExpensify

**Comment by trjExpensify** [comment]
I'm not sure when we'll take violations off the beta, @cead22 might have a better idea.

**Comment by shahinyan11** [comment]
> I'm not sure when we'll take violations off the beta, @cead22 might have a better idea.

What about [this](https://github.com/Expensify/App/issues/34418#issuecomment-1920773192) comment

**Comment by cead22** [comment]
We don't have a date for taking violations off beta, but we can add whatever logins you're using to the beta if you need to test, just share them with us

**Comment by cead22** [comment]
> If violations isn't getting taken off beta in the forseeable future then I'm not sure if [this design](https://github.com/Expensify/App/pull/36407#issuecomment-1943908250) is something that should be implemented in this PR.

This is already implemented

**Comment by trjExpensify** [comment]
Yeah, SmartScan errors aren't conditional on a violations beta either, so isn't it just taking the design pattern that Carlos references exists in the code and applying it to this case of a failed receipt scan that you edit only some of the fields required but still have violations? 

So these [here](https://github.com/Expensify/App/issues/34418#issuecomment-1920773192) would be displayed after the dot separator in the preview component.

**Comment by shahinyan11** [comment]
> So these [here](https://github.com/Expensify/App/issues/34418#issuecomment-1920773192) would be displayed after the dot separator in the preview component.

The current solution only displays `Report missing details` for all cases. will we have to change current solution

**Comment by trjExpensify** [comment]
>The current solution only displays Report missing details for all cases. will we have to change current solution

On a preview of an individual expense on a report? 😕

**Comment by shahinyan11** [comment]
> On a preview of an individual expense on a report? 😕

I do not understand your question well . I mean in the place for which this issue is open

**Comment by trjExpensify** [comment]
Got it, did you mean to say `Receipt missing details` rather than `Report missing details`?

**Comment by trjExpensify** [comment]
@cead22 do you agree with this for the money request preview after the dot separator? 

>Ok, so it should be as simple as appending the smartscan errors to the header text [message](https://github.com/Expensify/App/blob/12fc2cbae18ae6fbeda22bca5b8bcd2da63f5906/src/components/ReportActionItem/MoneyRequestPreview/MoneyRequestPreviewContent.tsx#L149).

**Comment by shahinyan11** [comment]
> Got it, did you mean to say `Receipt missing details` rather than `Report missing details`?

Ah, sorry. I wrote it wrong. I should be written `Receipt missing details`.

**Comment by cead22** [comment]
Can you be more specific
- `Receipt missing details` and the other variations are longer than 15 chars, does that mean we won't use the same logic we currently have for violations, where if the copy we're displaying is longer than 15 chars we show `Review required` instead?
- If not, would `Receipt missing details` take priority over violation messages or the other way around?

**Comment by shahinyan11** [comment]
> Can you be more specific
> 
> * `Receipt missing details` and the other variations are longer than 15 chars, does that mean we won't use the same logic we currently have for violations, where if the copy we're displaying is longer than 15 chars we show `Review required` instead?
> * If not, would `Receipt missing details` take priority over violation messages or the other way around?

@cead22 Who are these questions addressed to?

**Comment by trjExpensify** [comment]
Ah fair point about the 15 character limit logic. But yes, I think Olly's suggestion works then to drop the `receipt` portion.

**Comment by trjExpensify** [comment]
If @cead22 is happy, I'm happy. :)

**Comment by shahinyan11** [comment]
I'm waiting for someone's final approval to proceed.

**Comment by shahinyan11** [comment]
@cead22 I need some clarifying. What should be displayed if there are both smart scan errors and violations but the `violationMessage` is not long. Should not `Review required` be displayed somewhere in this case

**Comment by shahinyan11** [comment]
> @shahinyan11 In the case that there are both smart scan errors and violations we should display `Review required`.

`cash • Review required` or just `Review required` ?

**Comment by shahinyan11** [comment]
@Ollyws @cead22 Could you please take a look and confirm the behavior in video ?

https://github.com/Expensify/App/assets/46921547/9154d763-27a9-47dc-bb6e-b40be5b7b09c

**Comment by cead22** [comment]
Sorry for the delay, we had a small fire yesterday. Thanks for the video! 

One thing that looks odd to me is how it looks in the beginning, with both missing amount in large font, and missing merchant in smaller font, and also Review required
image

```
Cash ᐧ Review required
Missing amount
Missing merchant
```

@trjExpensify does that look right to you, or do you think the Review required in this case is redundant, and we should try something like
```
Cash
Missing amount
Missing merchant
```

**Comment by trjExpensify** [comment]
Yeah, I agree. I think including `Missing amount` and `Missing merchant` in the `amount` and `merchant` fields in the preview component is a bit strange -- because if you click into the expense that's not what you see on those fields at all.

I would just use the dot separator area to communicate what's missing, like:

Missing both `merchant` and `amount` (or "multiple violations" basically)

```
Cash ᐧ Review required   (dot) >
$0.00
```

Merchant entered, but missing amount:

```
Cash ᐧ Missing amount (dot) >
$0.00
The home depot 
```

Amount entered, but missing merchant:

```
Cash ᐧ Missing merchant  (dot) >
$10.00
```
CC: @dannymcclain @shawnborton

**Comment by shawnborton** [comment]
That sounds good to me Tom!

**Comment by shahinyan11** [comment]
@Ollyws @trjExpensify @shawnborton  I have updated cod and added new videos you can review

File: src/components/ReportActionItem/MoneyRequestPreview/MoneyRequestPreviewContent.tsx
Comment (shahinyan11) [reply src/components/ReportActionItem/MoneyRequestPreview/MoneyRequestPreviewContent.tsx:167]: @Ollyws We decided that the smart scan error should take priority. Doesn't current implementation match [this](https://github.com/Expensify/App/pull/36407#issuecomment-1969541669) comment

Comment (shahinyan11) [reply src/components/ReportActionItem/MoneyRequestPreview/MoneyRequestPreviewContent.tsx:167]: @Ollyws, @trjExpensify, @cead22, @shawnborton Can you give me a detailed specification for all possible cases . I need some detailed description to follow it

Comment (trjExpensify) [reply src/components/ReportActionItem/MoneyRequestPreview/MoneyRequestPreviewContent.tsx:167]: >In https://github.com/Expensify/App/pull/36407#issuecomment-1959954995 I specified that neither should take priority, so if there is a smartscan error and a violation then we should show Review required.

I agree with that. It's effectively "multiple violations" in that case.

Comment (shahinyan11) [reply src/components/ReportActionItem/MoneyRequestPreview/MoneyRequestPreviewContent.tsx:167]: @Ollyws @trjExpensify Sorry, but the information provided is not enough to understand. Can you explain what is the difference of https://github.com/Expensify/App/pull/36407#issuecomment-1959954995 and https://github.com/Expensify/App/pull/36407#issuecomment-1969541669 comments

Comment (shahinyan11) [reply src/components/ReportActionItem/MoneyRequestPreview/MoneyRequestPreviewContent.tsx:167]: @Ollyws It turns out it should be displayed there `Review required` If only merchant or only amount is missing and there is a violation errors . Is it correct ?

File: src/languages/es.ts
Comment (shahinyan11) [reply src/languages/es.ts:598]: I wrote about translations in slack https://expensify.slack.com/archives/C01GTK53T8Q/p1709035561391089 .

**Review by Ollyws — APPROVED** [review]
LGTM.

**Comment by techievivek** [comment]
I think that's fine and is certainly a backend issue.

**Review by techievivek — APPROVED** [review]

**Pull request merged**
